src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigManager.java

```diff
@@ -112,5 +123,43 @@ public Void call() throws Exception {
         }
 
     }
+    
+    static void reconcileJobsAndBuildConfigs() {
+        logger.info("Reconciling jobs and build configs");
+        List<WorkflowJob> jobs = Jenkins.getActiveInstance().getAllItems(WorkflowJob.class);
+        for (WorkflowJob job : jobs) {
+            BuildConfigProjectProperty property = job.getProperty(BuildConfigProjectProperty.class);
+            if (property != null) {
+                String ns = property.getNamespace();
+                String name = property.getName();
+                if (StringUtils.isNotBlank(ns) && StringUtils.isNotBlank(name)) {
+                    logger.info("Checking job " + job + " runs for BuildConfig " + ns + "/" + name);
+                    OpenShiftClient client = getAuthenticatedOpenShiftClient();
+                    BuildConfig bc = client.buildConfigs().inNamespace(ns).withName(name).get();
+                    if (bc == null) {
+                        try {
+                            ACL.impersonate(ACL.SYSTEM, new NotReallyRoleSensitiveCallable<Void, Exception>() {
+                                @Override
+                                public Void call() throws Exception {
+                                    try {
+                                        deleteInProgress(ns + name);
```

Comment:
the `deleteInProgress` methods takes as a signature the `BuildConfig` name. Is there any reason to prepend namespace ?

Reply:
I patterned off of https://github.com/openshift/jenkins-sync-plugin/blob/c6c5958a3f028112f24545a196f114342b2116a3/src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigManager.java#L94-L95 where both namespace and name are joined for the parameter supplied

also the corresponding deleteCompleted at https://github.com/openshift/jenkins-sync-plugin/blob/c6c5958a3f028112f24545a196f114342b2116a3/src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigManager.java#L100-L101 does that

finally, in general, we just need to account for the ns and name as 2 bc's in different namespaces could have the same name

Pretty sure this is right.

Reply:
ok, got it, I didn't check the other calls.
as an improvement, we should refactor this method then to set 2 parameters: `String namespace, String buildName`  or even better: juste the `BuildConfig bc` (as bc.metadata has reference to ns) and use a separator in the key map. 
Otherwise, you could face a crazy scenario if a user have this:
```
namespace1: java
bc1: build

namespace2: jav
bc2: abuild
```

Very poor likely to happen though.
I can push that RFE right after we merge this PR.